packages/graph/writers/unifi_writer.py

```
@@ -0,0 +1,852 @@
+
+    def __init__(self, neo4j_client: Neo4jClient, batch_size: int = 2000) -> None:
+        """Initialize UnifiWriter with Neo4j client.
+
+        Args:
+            neo4j_client: Neo4j client instance (must be connected).
+            batch_size: Number of rows per UNWIND batch (default 2000).
+        """
+        self.neo4j_client = neo4j_client
+        self.batch_size = batch_size
+
+        logger.info(f"Initialized UnifiWriter (batch_size={batch_size})")
```

Ruff (0.14.2)

Logging statement uses f-string

(G004)